jablib/src/test/java/org/jabref/logic/citation/repository/BibEntryCitationsAndReferencesRepositoryShellTest.java

```diff
@@ -90,14 +109,110 @@ void repositoryShouldWriteAndReadReferencesToAndFromExpectedDAO() {
                 new BibEntryRelationRepositoryMock(),
                 referencesDAO
         );
-        Assertions.assertFalse(bibEntryRelationsRepository.containsCitations(bibEntry));
+        Assertions.assertFalse(bibEntryRelationsRepository.containsReferences(bibEntry));
         Assertions.assertFalse(references.isEmpty());
 
         // WHEN
-        bibEntryRelationsRepository.insertCitations(bibEntry, references);
+        bibEntryRelationsRepository.insertReferences(bibEntry, references);
 
         // THEN
-        Assertions.assertTrue(bibEntryRelationsRepository.containsCitations(bibEntry));
-        Assertions.assertEquals(references, bibEntryRelationsRepository.readCitations(bibEntry));
+        Assertions.assertTrue(bibEntryRelationsRepository.containsReferences(bibEntry));
+        Assertions.assertEquals(references, bibEntryRelationsRepository.readReferences(bibEntry));
+    }
+
+    @Test
+    void readReferencesShouldReturnEmptyListWhenEntryIsNull() {
+        // GIVEN
+        BibEntryCitationsAndReferencesRepositoryShell repository = new BibEntryCitationsAndReferencesRepositoryShell(
+                new BibEntryRelationRepositoryMock(),
+                new BibEntryRelationRepositoryMock()
+        );
+
+        // WHEN
+        List<BibEntry> references = repository.readReferences(null);
```

Comment:
WHAT??????? NO!!!!!!

NO `null`. Work with `@Nullmarked` and `@Nullable` (which should not be placed any where)